server/models/comment.model.js

```diff
@@ -183,6 +196,22 @@ CommentSchema.statics = {
     return this.find({ parentComment: parentCommentId })
       .populate('author', '-password')
       .lean(); // so not Mongoose objects
+  },
+  upadteDeletedCommentContent(comments) {
+    if (comments) {
+      comments.forEach((comment) => {
+        this.upadteDeletedContent(comment);
+        // also updaate nested comment
+        comment.replies = this.upadteDeletedCommentContent(comment.replies); // eslint-disable-line
+      });
+    }
+    return comments;
+  },
+  upadteDeletedContent(comment) {
+    if (comment && comment.deleted) {
+      const deleteDate = moment(comment.dateDeleted).format('MM/DD/YYYY HH:mm:ss');
```

Comment:
Hey, just thinking this will actually error if comment.dateDeleted is null, which can happen for comments that have been deleted in the past (before this PR). We'll want to briefly document why we are doing this in the code I think i.e "Older comments didn't have the dateDeleted property".

We could probably just add a check and leave "Was deleted". 

Alternatively we can set an old date for all the existing deleted comments... but maybe this is for later and can add it as a task for the future so we can remove this annoying edge case. 

What do you think?

Reply:
Nice catch, didn't come to mind.

I'm inclined towards setting an old date so that we get it done with completely but would we then update the `dateDeleted` property? 🤔

Reply:
I am not sure exactly what you mean sorry?

Though after thinking about it, let's just add the if statement. It is pretty minor and we can add a simple comment why we are doing it for future reference.